resources/scripts/cite.js

```diff
@@ -0,0 +1,247 @@
+var locale;
+var citationData;
+var options = {
+  retrieveLocale: function () {
+    return locale;
+  },
+  retrieveItem: function (id) {
+    return citationData;
+  },
+};
+var cslDir = $('#cite-script').attr('src').split(/scripts\/.*\.js/).shift() + 'CSL/'
+var localeUrl = cslDir;
+var cslUrl = cslDir;
+var citations = {
+  chicago: '',
+  mla: '',
+  state: 'none',
+}
+var styleNames = {
+  chicago: 'chicago-fullnote-bibliography',
+  mla: 'modern-language-association',
+}
+var styles = {
+  chicago: '',
+  mla: '',
+}
+function baseCitation(id) {
+  return {
+    citationID: 'X',
+    citationItems: [
+      {
+        id: id,
+      }
+    ],
+    properties: {
+      noteIndex: 1
+    }
+  };
+}
+function updateState(state) {
+  console.log(citations.state, ' =>', state);
+  citations.state = state;
+}
+jQuery(function ($) {
+  function get(url, cb) {
+    $.get(url, function (data) {
+      cb(false, data);
+    }, 'text').fail(function () {
+      cb(true)
+    });
+  }
+  function getLocale(lang, cb) {
+    get(localeUrl + 'locale-' + lang + '.xml', cb);
+  }
+  function getCSL(style, cb) {
+    get(localeUrl + styleNames[style] + '.csl', cb);
+  }
+  function prepareCitations(lang, cb) {
+    try {
+      updateState('loading');
+      var alreadyFailed = false;
+      citationData = JSON.parse($('#original_citation').html())[0];
+      if (!locale) {
+        getLocale(lang, function (error, _locale) {
+          if (error) {
+            failed();
+          } else {
+            locale = _locale;
+            done();
+          }
+        });
+      }
+      if (!styles.chicago) {
+        getCSL('chicago', function (error, cslChicago) {
+          if (error) {
+            failed();
+          } else {
+            styles.chicago = cslChicago;
+            done();
+          }
+        });
+      }
+      if (!styles.mla) {
+        getCSL('mla', function (error, cslMla) {
+          if (error) {
+            failed();
+          } else {
+            styles.mla = cslMla;
+            done();
+          }
+        });
+      }
+      function done() {
+        if (locale && styles.chicago && styles.mla) {
+          updateState('ready');
+          cb(false);
+        }
+      }
+      function failed() {
+        if (!alreadyFailed) {
+          updateState('error');
+          cb(true);
+          alreadyFailed = true;
+        }
+      }
+      done();
+    } catch (error) {
+      console.error(error);
+      cb(true, error);
+    }
+  }
+  function generateCitation(style, cb) {
+    var citeproc = new CSL.Engine(options, styles[style]);
+    var citationResult = citeproc.processCitationCluster(baseCitation(citationData.id), [], []);
```

Comment:
Hi @ccheraa 

I think that we need to explicitly generate the bibliographic citation here with `makeBibliography` to ensure we get the correct MLA output.